synapse/notifier.py

```diff
@@ -93,6 +93,13 @@ def notify(self, notifier, events, start_token, end_token):
                 self.appservice, set()
             ).discard(self)
 
+        # Cancel the timeout for this notifer if one exists.
+        if self.timer is not None:
+            try:
+                notifier.clock.cancel_call_later(self.timer)
+            except:
+                logger.exception("Failed to cancel notifier timer")
```

Comment:
Will `cancel_call_later` no raise if the timer had already been cancelled? The timeout function invokes this method.